skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import operator
+from functools import wraps
+
+from ._intersection import IntervalTree
+from skbio.util._decorator import experimental
+
+
+class Interval:
+    """Stores the location and metadata of an interval feature.
```

Comment:
Even when you guys see the definition in the next sentence in the doc? I am
ok to change it, but is "interval" confusing to people? as it is usually
only mean a contiguous space between 2 numbers mathematically.

On Wed, Aug 31, 2016 at 2:22 PM, Jai Ram Rideout <notifications@github.com>
wrote:

> In skbio/metadata/_interval.py
> <https://github.com/biocore/scikit-bio/pull/1414#discussion_r77076020>:
>
> > +# Copyright (c) 2013--, scikit-bio development team.
> > +#
> > +# Distributed under the terms of the Modified BSD License.
> > +#
> > +# The full license is in the file COPYING.txt, distributed with this software.
> > +# ----------------------------------------------------------------------------
> > +
> > +import operator
> > +from functools import wraps
> > +
> > +from ._intersection import IntervalTree
> > +from skbio.util._decorator import experimental
> > +
> > +
> > +class Interval:
> > +    """Stores the location and metadata of an interval feature.
>
> I see, that is wordy. "interval feature" doesn't make sense to @ebolyen
> <https://github.com/ebolyen> and I though. What about just using
> "interval" instead of "interval feature"? The class is called Interval so
> it would be consistent with the idea it encompasses.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1414/files/12833254c244a61134eae4e690d2cf7cd363769f#r77076020>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA9CePIGTZA0Fyon-LRqMhNwqcaCbxFiks5qlfCRgaJpZM4JoIbD>
> .
>